### DISCUSSION TITLE
Feature Engineering vs Transformers

### DISCUSSION THREAD

**Discussion by BradKML** [alan-turing-institute/sktime#3524] [Q&A]
This repo contains tools for feature engineering for time series https://github.com/NITRO-AI/NitroFE
(for non-TS items there is something similar like [Feature Engine](https://github.com/feature-engine/feature_engine)) and [FeatureTools](https://github.com/alteryx/featuretools)
How are the transformers different from these libraries?

**Comment by fkiraly** [comment]
a (scikit-learn/sktime) transformer is the general concept and defines a unified software interface, based on a mathematical/conceptual abstraction.
The unified software interface allows to use them in pipelines and as components of other composite estimators, like grid-tuned pipelines.

The linked repositories have special operations/transformations in them (for feature extraction) which could be wrapped as sktime transformers, which in turn would enable their use in, say, time series forecasting pipelines.

So the relation between the linked feature extractors and sktime transformers is, on the conceptual level, "is an example of", and on the software level, "could be wrapped under the software interface specification to allow use as a pipeline component".

**Reply by fkiraly** [reply]
(feel free to open an issue of the kind "I would like to have wrapped [method X/library Y] in sktime interface")

**Reply by BradKML** [reply]
Just noticed that this also exists https://github.com/sktime/sktime/issues/2254
